File: app/src/main/java/com/pandatone/kumiwake/adapter/EditGroupListAdapter.java
package com.pandatone.kumiwake.adapter;

import android.content.Context;
import android.graphics.Color;
import android.text.InputFilter;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.TextView;

import com.pandatone.kumiwake.R;
import com.pandatone.kumiwake.kumiwake.KumiwakeCustom;
import com.pandatone.kumiwake.kumiwake.MainActivity;
import com.pandatone.kumiwake.member.GroupListAdapter;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Created by atsushi_2 on 2016/03/20.
 */
public class EditGroupListAdapter extends BaseAdapter {
    private Context context;
    private List<GroupListAdapter.Group> groupList;
    private static Map<Integer, EditText> groupNameView = new HashMap<Integer, EditText>();
    private static Map<Integer, EditText> memberNoView = new HashMap<Integer, EditText>();
    int beforeNo, afterNo;

    public EditGroupListAdapter(Context context, List<GroupListAdapter.Group> groupList) {
        this.context = context;
        this.groupList = groupList;
    }

    @Override
    public int getCount() {
        return groupList.size();
    }

    @Override
    public Object getItem(int position) {
        return groupList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }


    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        final EditText nameEditText, numberOfMemberEditText;
        TextView leader;
        View v = convertView;
        GroupListAdapter.Group listItem = groupList.get(position);
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (v == null) {
            v = inflater.inflate(R.layout.row_edit_group, null);
        }

        if (listItem != null) {
            nameEditText = (EditText) v.findViewById(R.id.editGroupName);
            numberOfMemberEditText = (EditText) v.findViewById(R.id.editTheNumberOfMember);
            leader = (TextView) v.findViewById(R.id.leader);
            nameEditText.setText(listItem.getGroup());
            numberOfMemberEditText.setText(String.valueOf(listItem.getBelongNo()));
            leader.setText(MainActivity.getContext().getText(R.string.leader) + ":" + MainActivity.getContext().getText(R.string.nothing));
            groupNameView.put(position, nameEditText);
            memberNoView.put(position, numberOfMemberEditText);
            numberOfMemberEditText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
                @Override
                public void onFocusChange(View v, boolean hasFocus) {
                    if (hasFocus) {
                        if (!numberOfMemberEditText.getText().toString().equals("") && !numberOfMemberEditText.getText().toString().equals("-")) {
                            KumiwakeCustom.scrollView.setOnTouchListener(new View.OnTouchListener() {
                                @Override
                                public boolean onTouch(View v, MotionEvent event) {
                                    return false;
                                }
                            });
                            beforeNo = Integer.parseInt(numberOfMemberEditText.getText().toString());
                        } else {
                            KumiwakeCustom.scrollView.setOnTouchListener(new View.OnTouchListener() {
                                @Override
                                public boolean onTouch(View v, MotionEvent event) {
                                    return true;
                                }
                            });
                        }
                        numberOfMemberEditText.setFilters(new InputFilter[]{new InputFilter.LengthFilter(2)});
                    } else {

                        if (!numberOfMemberEditText.getText().toString().equals("") && !numberOfMemberEditText.getText().toString().equals("-")) {
                            KumiwakeCustom.scrollView.setOnTouchListener(new View.OnTouchListener() {
                                @Override
                                public boolean onTouch(View v, MotionEvent event) {
                                    return false;
                                }
                            });

                                afterNo = Integer.parseInt(numberOfMemberEditText.getText().toString());

                            int addNo = beforeNo - afterNo;
                            KumiwakeCustom.changeBelongNo(position, addNo);
                            if (afterNo < 0) {
                                numberOfMemberEditText.setTextColor(Color.RED);
                            } else {
                                numberOfMemberEditText.setTextColor(Color.BLACK);
                            }
                        } else {
                            KumiwakeCustom.scrollView.setOnTouchListener(new View.OnTouchListener() {
                                @Override
                                public boolean onTouch(View v, MotionEvent event) {
                                    return true;
                                }
                            });
                            afterNo = 0;
                        }
                    }
                }
            });

        }

        return v;
    }

    public static String getGroupName(int position) {
        String groupName = MainActivity.getContext().getText(R.string.nothing).toString();
        EditText groupNameEditText = groupNameView.get(position);
        if (groupNameEditText.getText() != null) {
            groupName = groupNameEditText.getText().toString();
        }
        return groupName;
    }

    public static int getMemberNo(int position) {
        int memberNo = 0;
        EditText memberNoEditText = memberNoView.get(position);
        if (memberNoEditText.getText().toString().length() > 0) {
            memberNo = Integer.parseInt(memberNoEditText.getText().toString());
        }
        return memberNo;
    }


    public static void setRowHeight(ListView listView, EditGroupListAdapter listAdapter) {
        int totalHeight = 0;

        for (int j = 0; j < listAdapter.getCount(); j++) {
            View item = listAdapter.getView(j, null, listView);
            item.measure(0, 0);
            totalHeight += item.getMeasuredHeight();
        }

        listView.getLayoutParams().height = totalHeight + (listView.getDividerHeight() * (listAdapter.getCount() - 1));
        listView.requestLayout();
    }

}

File: app/src/main/java/com/pandatone/kumiwake/adapter/EditGroupListAdapter.kt
package com.pandatone.kumiwake.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.text.InputFilter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.pandatone.kumiwake.R
import com.pandatone.kumiwake.kumiwake.KumiwakeCustom
import com.pandatone.kumiwake.member.Group
import java.util.*

/**
 * Created by atsushi_2 on 2016/03/20.
 */
class EditGroupListAdapter(private val context: Context, private val groupList: List<Group>, private val scrollView: ScrollView) : BaseAdapter() {
    private var beforeNo: Int = 0
    private var afterNo: Int = 0

    override fun getCount(): Int {
        return groupList.size
    }

    override fun getItem(position: Int): Any {
        return groupList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }


    @SuppressLint("ClickableViewAccessibility", "SetTextI18n", "InflateParams")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        val nameEditText: EditText
        val numberOfMemberEditText: EditText
        val leader: TextView
        var v = convertView
        val group = groupList[position]
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        if (v == null) {
            v = inflater.inflate(R.layout.row_edit_group, null)
        }

        nameEditText = v!!.findViewById<View>(R.id.editGroupName) as EditText
        numberOfMemberEditText = v.findViewById<View>(R.id.editTheNumberOfMember) as EditText
        leader = v.findViewById<View>(R.id.leader) as TextView
        nameEditText.setText(group.name)
        numberOfMemberEditText.setText(group.belongNo.toString())
        leader.text = "${context.getString(R.string.leader)} : ${context.getString(R.string.nothing)}"
        groupNameView[position] = nameEditText
        memberNoView[position] = numberOfMemberEditText
        numberOfMemberEditText.onFocusChangeListener = View.OnFocusChangeListener { _, hasFocus ->
            if (hasFocus) {
                if (numberOfMemberEditText.text.toString() != "" && numberOfMemberEditText.text.toString() != "-") {
                    scrollView.setOnTouchListener { _, _ -> false }
                    beforeNo = Integer.parseInt(numberOfMemberEditText.text.toString())
                } else {
                    scrollView.setOnTouchListener { _, _ -> true }
                }
                numberOfMemberEditText.filters = arrayOf<InputFilter>(InputFilter.LengthFilter(2))
            } else {

                if (numberOfMemberEditText.text.toString() != "" && numberOfMemberEditText.text.toString() != "-") {
                    scrollView.setOnTouchListener { _, _ -> false }

                    afterNo = Integer.parseInt(numberOfMemberEditText.text.toString())

                    val addNo = beforeNo - afterNo
                    KumiwakeCustom().changeBelongNo(position, addNo)
                    if (afterNo < 0) {
                        numberOfMemberEditText.setTextColor(Color.RED)
                    } else {
                        numberOfMemberEditText.setTextColor(Color.BLACK)
                    }
                } else {
                    scrollView.setOnTouchListener { _, _ -> true }
                    afterNo = 0
                }
            }
        }

        return v
    }

    companion object {
        @SuppressLint("UseSparseArrays")
        private val groupNameView = HashMap<Int, EditText>()
        @SuppressLint("UseSparseArrays")
        private val memberNoView = HashMap<Int, EditText>()

        fun getGroupName(position: Int): String {
            var groupName = R.string.nothing.toString()
            val groupNameEditText = groupNameView[position]
            if (groupNameEditText!!.text != null) {
                groupName = groupNameEditText.text.toString()
            }
            return groupName
        }

        fun getMemberNo(position: Int): Int {
            var memberNo = 0
            val memberNoEditText = memberNoView[position]
            if (memberNoEditText!!.text.toString().isNotEmpty()) {
                memberNo = Integer.parseInt(memberNoEditText.text.toString())
            }
            return memberNo
        }


        fun setRowHeight(listView: ListView, listAdp: EditGroupListAdapter) {
            var totalHeight = 0

            for (j in 0 until listAdp.count) {
                val item = listAdp.getView(j, null, listView)
                item?.measure(0, 0)
                totalHeight += item!!.measuredHeight
            }

            listView.layoutParams.height = totalHeight + listView.dividerHeight * (listAdp.count - 1)
            listView.requestLayout()
        }
    }

}

